src/Symfony/Component/Messenger/Transport/RedisExt/Connection.php

```diff
@@ -113,23 +117,51 @@ public function get(): ?array
 
     public function ack(string $id): void
     {
-        $this->connection->xack($this->stream, $this->group, [$id]);
+        $e = null;
+        try {
+            $acknowledged = $this->connection->xAck($this->stream, $this->group, [$id]);
+        } catch (\RedisException $e) {
+        }
+
+        if (!$acknowledged || $e) {
+            throw new TransportException(($e ? $e->getMessage() : $this->connection->getLastError()) ?? sprintf('Could not acknowledge redis message "%s".', $id), 0, $e);
+        }
     }
 
     public function reject(string $id): void
     {
-        $this->connection->xdel($this->stream, [$id]);
+        $e = null;
+        try {
+            $deleted = $this->connection->xDel($this->stream, [$id]);
+        } catch (\RedisException $e) {
+        }
+
+        if (!$deleted || $e) {
+            throw new TransportException(($e ? $e->getMessage() : $this->connection->getLastError()) ?? sprintf('Could not delete message "%s" from the redis stream.', $id), 0, $e);
+        }
     }
 
     public function add(string $body, array $headers)
     {
-        $this->connection->xadd($this->stream, '*', ['message' => json_encode(
-            ['body' => $body, 'headers' => $headers]
-        )]);
+        $e = null;
+        try {
+            $added = $this->connection->xAdd($this->stream, '*', ['message' => json_encode(
+                ['body' => $body, 'headers' => $headers]
+            )]);
+        } catch (\RedisException $e) {
+        }
+
+        if (!$added || $e) {
+            throw new TransportException(($e ? $e->getMessage() : $this->connection->getLastError()) ?? 'Could not add a message to the redis stream.', 0, $e);
+        }
     }
 
     public function setup(): void
     {
-        $this->connection->xgroup('CREATE', $this->stream, $this->group, 0, true);
+        try {
+            $this->connection->xGroup('CREATE', $this->stream, $this->group, 0, true);
```

Comment:
@chalasr Thanks for the fixes. Just to mention it here if mocking this with prophecy the function need to be called the lowercase version. See https://github.com/phpredis/phpredis/issues/1489

Reply:
Thanks! see #31346